src/App.tsx

```diff
@@ -15,37 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'name',
+  Length = 'length',
+}
+
+interface SortOptions {
+  sortField: string;
+  reverse: boolean;
+}
+
+interface GoodListProps {
+  goods: string[];
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: SortOptions,
+): string[] {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const GoodList: React.FC<GoodListProps> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li key={good} data-cy="Good">
+        {good}
+      </li>
+    ))}
+  </ul>
+);
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
```

Comment:
The initial state of `sortField` is set to an empty string. This might cause unexpected behavior in the `getPreparedGoods` function since it expects a valid `sortField` to perform sorting. Consider initializing `sortField` with a default value from the `SortType` enum, such as `SortType.Alphabet` or `SortType.Length`, to ensure consistent behavior.